src/controller/CHIPDeviceController.cpp

```diff
@@ -2118,7 +2118,18 @@ void DeviceCommissioner::PerformCommissioningStep(DeviceProxy * proxy, Commissio
             CommissioningStageComplete(err);
             return;
         }
-        err = proxy->SetPeerId(params.GetRootCert().Value(), params.GetNoc().Value());
+
+        FabricId fabricId;
+        NodeId nodeId;
+        err = ExtractNodeIdFabricIdFromOpCert(params.GetNoc().Value(), &nodeId, &fabricId);
+        if (err != CHIP_NO_ERROR)
+        {
+            ChipLogError(Controller, "Error extracting nodeid from cert: %s", err.AsString());
+            CommissioningStageComplete(err);
+            return;
+        }
+
+        err = proxy->SetNodeId(GetPeerScopedId(nodeId));
```

Comment:
This implies the commissioner has a fabric table, but it may not.  It is possible to install credentials in the commissionee over PASE, but delegate CASE to a different node.  We don't really know for certain that the commissioner has a fabric table until such time as the commissioner is itself attempting CASE.

Reply:
Makes sense. That being said, is anyone actually using the nodeId stored within the `CommissioneeDeviceProxy` object? I don't see any calls to `GetPeerId` from within or outside that class? If so, can we just delete it?

Reply:
There are calls to `GetDeviceId`.  But those only care about the node id, not the fabric bits.

Reply:
Alternatively, `CommissionableDeviceProxy` could store the fully-qualified identity, <root CA public key, fabric ID, node ID>.  It's 72 bytes.

In defense of what was here previously, the PASE phase of commissioning on the commissioner side is the one instance I can think of where it is completely legitimate not to have a fabric table.  Using <compressed fabric ID, node ID> in the `CommissionableDeviceProxy` accommodates that subtlety.

Reply:
K, will put it back.